File: memegenerator/src/main/java/com/innov8/memegenerator/custom_views/ToggleImageButton.java
package com.innov8.memegenerator.custom_views;

import android.content.Context;
import android.content.res.TypedArray;
import android.graphics.Color;
import android.support.v7.widget.AppCompatImageButton;
import android.util.AttributeSet;

import com.innov8.memegenerator.R;

public class ToggleImageButton extends AppCompatImageButton {
    private boolean checked;
    private int checkedbackground;
    private int uncheckedbackground;


    public ToggleImageButton(Context context) {
        super(context);
        checkedbackground=Color.parseColor("#5555");
        uncheckedbackground=Color.TRANSPARENT;
        setChecked(false);
    }

    public ToggleImageButton(Context context, AttributeSet attrs) {
        super(context, attrs);
        init(attrs);
    }

    public ToggleImageButton(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init(attrs);
    }

    private void init(AttributeSet attrs){
        TypedArray a = getContext().getTheme().obtainStyledAttributes(
                attrs,
                R.styleable.ToggleImageButton,
                0, 0);

        try {
            checked=a.getBoolean(R.styleable.ToggleImageButton_checked,false);
           checkedbackground=a.getColor(R.styleable.ToggleImageButton_checked_background,  Color.parseColor("#5555"));
           uncheckedbackground=a.getColor(R.styleable.ToggleImageButton_checked_background, Color.TRANSPARENT);
        } finally {
            a.recycle();
        }
        setChecked(checked);
    }

    public boolean isChecked() {
        return checked;
    }

    public void setChecked(boolean checked) {
        this.checked = checked;
        if (checked){
            setBackgroundColor(checkedbackground);
        }else{
            setBackgroundColor(uncheckedbackground);
        }
    }
}

File: memegenerator/src/main/java/com/innov8/memegenerator/customViews/ToggleImageButton.kt
package com.innov8.memegenerator.customViews

import android.content.Context
import android.graphics.Color
import androidx.appcompat.widget.AppCompatImageButton
import android.util.AttributeSet
import com.innov8.memegenerator.R

class ToggleImageButton : AppCompatImageButton {
    var isChecked: Boolean = false
        private set
    private var checkedColor: Int = Color.RED
    private var uncheckedColor: Int = Color.WHITE

    var onCheckChanged:((checked:Boolean,fromUser:Boolean)->Unit)?=null
    constructor(context: Context) : super(context) {
        init(null)
    }

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {
        init(attrs)
    }

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {
        init(attrs)
    }

    private fun init(attrs: AttributeSet?) {
        if(attrs!=null){
            val a = context.theme.obtainStyledAttributes(
                    attrs,
                    R.styleable.ToggleImageButton,
                    0, 0)
            try {
                isChecked = a.getBoolean(R.styleable.ToggleImageButton_checked, false)
                checkedColor = a.getColor(R.styleable.ToggleImageButton_checked_background, Color.RED)
                uncheckedColor = a.getColor(R.styleable.ToggleImageButton_unchecked_background, Color.WHITE)
            } finally {
                a.recycle()
            }
        }

        setChecked(isChecked, false)
        setOnClickListener {
            toggleCheck(true)
        }
        setBackgroundColor(Color.TRANSPARENT)
    }
    fun setChecked(checked: Boolean, fromUser: Boolean=false) {
        this.isChecked = checked
        if (checked) {
            setColorFilter(checkedColor)
        } else {
            setColorFilter(uncheckedColor)
        }
        onCheckChanged?.invoke(checked,fromUser)
    }
    fun toggleCheck(fromUser: Boolean=false) {
        setChecked(!isChecked, fromUser)
    }
}
